pkg/virt-manifest/defaults.go

```diff
@@ -0,0 +1,56 @@
+package virt_manifest
+
+import (
+	"fmt"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+// AddMinimalDomainSpec takes an existing domain spec and ensures
+// That libvirt's minimum required values are assigned a default
+func AddMinimalDomainSpec(domain *v1.DomainSpec) error {
+	if domain.Name == "" {
+		return fmt.Errorf("domain name cannot be empty")
+	}
+
+	if domain.Type == "" {
+		domain.Type = "qemu"
+	}
+
+	if domain.OS.Type.OS == "" {
+		domain.OS.Type.OS = "hvm"
+	}
+
+	if domain.Memory.Unit == "" {
+		domain.Memory.Unit = "KiB"
```

Comment:
We can't guess that, we should make the memory related fields mandatory fields ...